turbinia/lib/utils.py

```diff
@@ -22,35 +22,58 @@
 import tempfile
 import threading
 
+from turbinia import config
 from turbinia import TurbiniaException
 
 log = logging.getLogger('turbinia')
 
 DEFAULT_TIMEOUT = 7200
 
 
-def _image_export(command, output_dir, timeout=DEFAULT_TIMEOUT):
+def _image_export(command, output_dir, disk_path, timeout=DEFAULT_TIMEOUT):
   """Runs image_export command.
 
   Args:
     command: image_export command to run.
     output_dir: Path to directory to store the extracted files.
+    disk_path: Path to either a raw disk image or a block device.
 
   Returns:
     list: paths to extracted files.
 
   Raises:
     TurbiniaException: If an error occurs when running image_export.
   """
-  # TODO: Consider using the exec helper to gather stdin/err.
-  log.debug(f"Running image_export as [{' '.join(command):s}]")
-  try:
-    subprocess.check_call(command, timeout=timeout)
-  except subprocess.CalledProcessError as exception:
-    raise TurbiniaException(f'image_export.py failed: {exception!s}')
-  except subprocess.TimeoutExpired as exception:
-    raise TurbiniaException(
-        f'image_export.py timed out after {timeout:d}s: {exception!s}')
+
+  # Execute the job via docker if docker is enabled and the Job has an image configured
+  # The image should be of the log2timeline/plaso type.
+  config.LoadConfig()
+  dependencies = config.ParseDependencies()
+  docker_image = None
+  job_name = 'FileArtifactExtractionJob'.lower()
+  if dependencies.get(job_name):
+    docker_image = dependencies.get(job_name).get('docker_image')
+  if (config.DOCKER_ENABLED and docker_image is not None):
+    from turbinia.lib import docker_manager
+    ro_paths = [disk_path]
+    rw_paths = [output_dir]
+    container_manager = docker_manager.ContainerManager(docker_image)
+    log.info(
+        'Executing job {0:s} in container: {1:s}'.format(
+            job_name, docker_image))
+    stdout, stderr, ret = container_manager.execute_container(
+        command, shell=False, ro_paths=ro_paths, rw_paths=rw_paths,
+        timeout_limit=timeout)
```

Comment:
I wonder if we should use the job timeout here instead of the default timeout?

Reply:
Good point. Changed to job_timeout value unless there is no job timeout set in the configuration file for that job it will default to the default timeout.